src/stereo.cpp

```diff
@@ -11,15 +11,39 @@
 const char* VERTEX_SOURCE = R"(
 #version 330 core
 
+#define M_PI 3.141592653589793238462643383279502884197169399375105820974
+
 layout (location = 0) in vec3 position;
 layout (location = 1) in vec4 color;
 layout (location = 2) in vec2 texture_coord;
 
 out vec4 frag_color;
 out vec2 frag_texture_coord;
 
+uniform float time;
+
+vec3 axis_rotate(vec3 axis, vec3 coord, float angle) {
+  float rot_cos = cos(angle), rot_sin = sin(angle);
+  float inv_cos = (1 - rot_cos);
+  mat3 rotate = mat3(
+    vec3(
+      rot_cos + axis.x*axis.x*inv_cos,
+      axis.y*axis.x*inv_cos + axis.z*rot_sin,
+      axis.z*axis.x*inv_cos - axis.y*rot_sin),
+    vec3(
+      axis.x*axis.y*inv_cos - axis.z*rot_sin,
+      rot_cos + axis.y*axis.y*inv_cos,
+      axis.z*axis.y*inv_cos + axis.x*rot_sin),
+    vec3(
+      axis.x*axis.z*inv_cos + axis.y*rot_sin,
+      axis.y*axis.z*inv_cos - axis.x*rot_sin,
+      rot_cos + axis.z*axis.z*inv_cos));
+  return rotate * coord;
+}
```

Comment:
Since this code is quite general and in principle reusable, we might want to move it (and other linear algebra snippets) into a separate file eventually.